src/core/server/http/router/validator/validator.ts

```diff
@@ -170,23 +170,23 @@ export class RouteValidator<P = {}, Q = {}, B = {}> {
    * @internal
    */
   public getParams(data: unknown, namespace?: string): Readonly<P> {
-    return this.validate(this.config.params, this.options.unsafe?.params, data, namespace);
+    return this.validate(this.config.params, this.options.unsafe?.params, data, namespace) as P;
```

Comment:
So, introducing this change in the `ObjectResultType` caused errors due to `RouteValidationSpec` and `RouteValidationResultType` types being different for `ObjectType` schemas, as the keys mutates. However after spending a lot of time on that (and with @afharo 's help), it appear that this can't really be fixed without a reverse type for `ObjectResultType` (something like `ObjectParamsFromResultType`), to properly fix `RouteValidationResultType`, which is not technically doable. 

I think this is still fine because:
- type inference is still working, and the `maybe` property keys are correctly considered optionals in the resulting `request` properties when using in route handlers. It's really only the internal/private methods typing that needs this fix (the return type of `validate`). it 'works as expected' for a public API point of view.
- we got good test coverage for this code.

But if someone wants to take another look, please do 😄